mods/dungeon_loot/mapgen.lua

```diff
@@ -0,0 +1,159 @@
+minetest.set_gen_notify({dungeon=true})
+
+local function noise3d_integer(noise, pos)
+	return math.abs(math.floor(noise:get3d(pos) * 2147483647))
+end
+
+local function random_sample(rand, list, count)
+	local ret = {}
+	for n = 1, count do
+		local idx = rand:next(1, #list)
+		table.insert(ret, list[idx])
+		table.remove(list, idx)
+	end
+	return ret
+end
+
+local function find_walls(cpos)
+	local wall = minetest.registered_aliases["mapgen_cobble"]
+	local wall_alt = minetest.registered_aliases["mapgen_mossycobble"]
+	local wall_sand = minetest.registered_aliases["mapgen_sandstonebrick"]
+	local is_wall = function(node)
+		return table.indexof({wall, wall_alt, wall_sand}, node.name) ~= -1
+	end
+
+	local dirs = { {x=1, z=0}, {x=-1, z=0}, {x=0, z=1}, {x=0, z=-1} }
+	local get_node = minetest.get_node
+
+	local ret = {}
+	local mindist = {x=0, z=0}
+	local min = function(a, b) return a ~= 0 and math.min(a, b) or b end
+	local wallnode
+	for _, dir in ipairs(dirs) do
+		for i = 1, 8 do -- 8 = max room size / 2
```

Comment:
Dungeon max room size is 18x18 so this can be 9.
https://github.com/minetest/minetest/blob/master/src/mapgen.cpp#L891